src/components/lib/cos_kernel_api.c

```diff
@@ -649,6 +649,10 @@ int
 cos_thd_switch(thdcap_t c)
 { return call_cap_op(c, 0, 0, 0, 0, 0); }
 
+int
+cos_thd_wakeup(thdcap_t thd, tcap_t tc, tcap_prio_t prio)
+{ return call_cap_op(tc, CAPTBL_OP_TCAP_WAKEUP, thd, (prio << 32) >> 32, prio >> 32, 0); }
```

Comment:
nit: use mask to clear the higher 32 bits. It should be a pretty common operation so a macro is probably necessary.